crates/rpc/tests/rpc.rs

```diff
@@ -0,0 +1,483 @@
+// These tests need Beerus to run in the background, hence why they're hidden behind the following feature.
+#![cfg(feature = "integration-tests")]
+
+use reqwest::Url;
+use starknet::{
+    core::types::{
+        BlockId, BlockTag, BlockWithTxHashes, BlockWithTxs, DeclareTransaction,
+        DeployAccountTransaction, Event, EventFilter, FieldElement,
+        InvokeTransaction, MaybePendingBlockWithTxHashes,
+        MaybePendingBlockWithTxs, MaybePendingTransactionReceipt, Transaction,
+        TransactionReceipt,
+    },
+    providers::{
+        jsonrpc::HttpTransport, JsonRpcClient, Provider, ProviderError,
+    },
+};
+
+struct TestContext<T> {
+    client: JsonRpcClient<HttpTransport>,
+    block: BlockWithTxs,
+    block_id: BlockId,
+    extracted_value: T,
+}
+
+/// Creates a `TestContext` with the latest block.
+async fn latest_block_context() -> TestContext<()> {
+    context(|_block| Some(())).await
+}
+
+/// Creates a `TestContext` with the latest block with at least 10 transactions.
+async fn latest_10_tx_context() -> TestContext<()> {
```

Comment:
nit: maybe we could use transactions count as a parameter, so we can find a block with any number or transactions if we ever need more or less, or is there a reason it is 10?

Reply:
10 is arbitrary. Checking all the transactions can take a long time and result in thousands of calls, so this limit is purely practical. But you are right on the name & parameter, this is inelegant.